packages/react-router/src/components/FocusContextProvider/FocusContextProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import React, {useRef, useEffect, ReactNode} from 'react';
+import {FocusContext} from '../../context';
+
+export function FocusContextProvider({children}: {children: ReactNode}) {
+  const focusRef = React.createRef<HTMLElement>();
+  const focus = () => {
+    return focusRef?.current?.focus();
+  };
+
+  const firstRender = useRef(true);
+
+  useEffect(() => {
+    if (!firstRender.current) {
+      focus();
+    }
+  });
+
+  useEffect(() => {
+    firstRender.current = false;
```

Comment:
I'd put this in the first effect, since the check just needs to know to go from `true` => `false` (and not do some work if it started `true`)